src/cli/vite.js

```diff
@@ -29,55 +29,68 @@ function getResolvedAliases(dir) {
  * @param {import('../plugin/index').SlinkityConfigOptions['dir']} dir
  * @returns {import('vite').UserConfigExport}
  */
-async function getSharedConfig(dir) {
-  const importAliasesToResolvedPath = Object.entries(getResolvedAliases(dir)).map(
+async function getSharedConfig(eleventyDir) {
+  const importAliasesToResolvedPath = Object.entries(getResolvedAliases(eleventyDir)).map(
     ([key, value]) => [IMPORT_ALIASES[key], value],
   )
-  let optimizeDeps = {}
+  let reactConfig = vite.defineConfig()
   try {
     require('react')
     require('react-dom')
-    optimizeDeps = {
-      include: ['react', 'react/jsx-runtime', 'react/jsx-dev-runtime', 'react-dom'],
-    }
+    reactConfig = vite.defineConfig({
+      resolve: {
+        dedupe: ['react', 'react-dom'],
+      },
+      optimizeDeps: {
+        include: ['react', 'react/jsx-runtime', 'react/jsx-dev-runtime', 'react-dom'],
+      },
+      build: {
+        rollupOptions: {
+          output: {
+            manualChunks: {
+              react: ['react'],
+            },
+          },
+        },
+      },
+    })
   } catch {
     // no-op
   }
-  return vite.defineConfig({
-    clearScreen: false,
-    configFile: await getConfigFile(),
-    resolve: {
-      alias: Object.fromEntries(importAliasesToResolvedPath),
-      dedupe: ['react', 'react-dom'],
-    },
-    optimizeDeps,
-  })
+  return vite.mergeConfig(
+    vite.defineConfig({
+      clearScreen: false,
+      configFile: await getConfigFile(),
+      resolve: {
+        alias: Object.fromEntries(importAliasesToResolvedPath),
+      },
+    }),
+    reactConfig,
+  )
 }
 
 /**
  * Build production bundle
  * @param {import('../plugin/index').SlinkityConfigOptions['dir']} dir
  */
-async function build(dir) {
-  const inputFiles = await glob(`${dir.input}/**/*.html`, { absolute: true })
-
+async function build({ eleventyDir, input, output }) {
+  const inputFiles = await glob(`${input}/**/*.html`, { absolute: true })
   if (inputFiles.length) {
-    await vite.build({
-      ...(await getSharedConfig(dir)),
-      root: dir.input,
-      build: {
-        outDir: dir.output,
-        emptyOutDir: true,
-        rollupOptions: {
-          input: inputFiles,
-          output: {
-            manualChunks: {
-              react: ['react'],
+    await vite.build(
+      vite.mergeConfig(
```

Comment:
Learned about Vite's `mergeConfig` helper recently. Want to refactor all `...` spreads across our configs when we find them!